misc/demo_call_cpp.py

```diff
@@ -0,0 +1,34 @@
+import ctypes
+import os
+
+import taichi as ti
+
+ti.init(dynamic_index=True)
+
+source = '''
+void foo(int* a, int b[2][2]) {
+    a[0] = 11;
+    b[0][0] = 10;
+}
+'''
+
+with open('a.c', 'w') as f:
+    f.write(source)
+
+# TODO: Integrate this setup into taichi
+os.system("clang-11 -flto -c a.c -o a.bc")
```

Comment:
We had a discussion offline, so  don't expose this to the users. Create a small class that wraps up things like `type, so_func,  asm_source, llvm_module`, which also takes care of creating/deleting these temp files. This way we also decouples the loading of the LLVM module from the codegen itself